### PR TITLE
WooPay naming/styling updates

### PR DESCRIPTION
**Pull request by malithsen**
Fixes 635-gh-Automattic/woopay

#### Changes proposed in this Pull Request

* Update registration update copy (see 635-gh-Automattic/woopay)
* Replace Platform Checkout with WooPay in the UI
* Replace the placeholder Woo logo on the settings page to use the WooPay logo
* Use production URLs as the default in `PLATFORM_CHECKOUT_FRONTEND_HOST` and `PLATFORM_CHECKOUT_HOST`

Context: p1649353944790879-slack-C022WMN88KG, pdpOw2-yo-p2

#### Testing instructions

* Make sure platform checkout is enabled for your merchant site.
* Ensure that WooPay registration UI matches the updated design in pdpOw2-yo-p2. Note that the link in "powered by WooPay" is not set on this PR, this will be set later.
* When `PLATFORM_CHECKOUT_FRONTEND_HOST` and `PLATFORM_CHECKOUT_HOST` is not set, WCPay attempts to load the OTP iframe from `woo.app`
* WooCommerce -> Settings -> Payments -> Express Checkout page uses the correct logo cO37XZJMyjQweZ6RhEeDhX-fi-31%3A25535

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) : _Add link here / 'QA Testing Not Applicable'_

### CODE REVIEW

File: client/components/platform-checkout/save-user/agreement.js [hunk 1]
```diff
@@ -10,24 +10,24 @@ const Agreement = () => {
...
 						>
 							{ __( 'Terms of Service', 'woocommerce-payments' ) }
 						</a>
 					),
 					privacyPolicy: (
 						<a
-							href="https://automattic.com/privacy/"
 							target="_blank"
-							rel="noreferrer"
+							href="https://href.li/?https://automattic.com/privacy/"
```
Comment (bborman22) [inline client/components/platform-checkout/save-user/agreement.js:29]: I think this might just be a copy/paste artifact to address?

**Review by bborman22 — CHANGES_REQUESTED** [review]
This all looked good! I found one small typo and one link issue, but otherwise everything matched the designs and descriptions I saw in discussion and designs.

I did see that we reference platform checkout in the [payment settings API parameter descriptions](https://github.com/Automattic/woocommerce-payments/blob/56595f45b9e083c2d3e2374c7e528199267b7980/includes/admin/class-wc-rest-payments-settings-controller.php#L202). I for sure don't feel that's a high priority to address (and maybe shouldn't even be changed?) and should not block anything, but figured I'd mention it in case this would be a good time to update that?

After the link and typo are addressed this LGTM!

**Review by bborman22 — APPROVED** [review]
Thanks for those fixes, LGTM!

**Comment by malithsen** [comment]
Thanks for the review @bborman22, fixed the link and typo in https://github.com/Automattic/woocommerce-payments/pull/4107/commits/f8532c5d359aeacb1fad6c41a457db88537ebbf1

> I did see that we reference platform checkout in the [payment settings API parameter descriptions](https://github.com/Automattic/woocommerce-payments/blob/56595f45b9e083c2d3e2374c7e528199267b7980/includes/admin/class-wc-rest-payments-settings-controller.php#L202). I for sure don't feel that's a high priority to address (and maybe shouldn't even be changed?)

I missed this initially, I adjusted the description in https://github.com/Automattic/woocommerce-payments/pull/4107/commits/a42d5a1928062f534fe197ea0862fc6656c3a8da but I am not sure if it ever shows up in the UI or where it shows up.

File: client/components/platform-checkout/save-user/agreement.js [hunk 2]
```diff
@@ -10,24 +10,24 @@ const Agreement = () => {
 		<div className="tos">
 			{ interpolateComponents( {
 				mixedString: __(
-					'By entering your phone number and completing your purchase, you will create a Platform Checkout account and agree to {{termsOfService/}} and {{privacyPolicy/}}.',
+					'By entering your phone number and completing your purchase, you will create a WooPay account and agree to {{termsOfService/}} and {{privacyPolicy/}}.',
```
Comment (leonardola) [inline client/components/platform-checkout/save-user/agreement.js:13]: this should read:

By entering your mobile phone number and completing your purchase, you will create a WooPay account, and you agree to the Terms of Service and Privacy Policy

**Review by leonardola — CHANGES_REQUESTED** [review]
@malithsen I noticed that the checkbox uses the default browser styling while the Updated design uses a WooCommerce color.
How it actually looks:
Screen Shot 2022-04-08 at 2 16 47 PM

How it should look:
Screen Shot 2022-04-08 at 2 19 01 PM

Also the label "Mobile phone number" is missing before the phone number input.

How it actually looks:
Screen Shot 2022-04-08 at 2 19 46 PM


How it should look:
Screen Shot 2022-04-08 at 2 19 51 PM
 
I also left a comment about a string that is different from the one levin posted

File: client/components/platform-checkout/save-user/agreement.js (hunk 2 above)
Comment (malithsen) [reply client/components/platform-checkout/save-user/agreement.js]: Updated in https://github.com/Automattic/woocommerce-payments/pull/4107/commits/b3da826f56743af9f316ba7a8d5c8b05f2e95664

**Comment by malithsen** [comment]
> Also the label "Mobile phone number" is missing before the phone number input.

Nice catch! Updated in https://github.com/Automattic/woocommerce-payments/pull/4107/commits/49a5539daef82a64c1f7d8cc9f94002e7598ebaa

> I noticed that the checkbox uses the default browser styling while the Updated design uses a WooCommerce color.

I played this for a bit but couldn't get it exactly right. I'd like to separate this out to a different issue to unblock the merge of the rest of the changes here.

**Review by leonardola — APPROVED** [review]
This looks perfect now. Thanks for working on this one. :shipit:

**Pull request merged**